mxfusion/models/factor_graph.py

```diff
@@ -461,36 +467,40 @@ def reconcile_graphs(current_graphs, primary_previous_graph, secondary_previous_
 
         :rtype: {previous ModelComponent : current ModelComponent}
         """
+
+        def update_with_named_components(previous_components, current_components, component_map, current_level):
+            for c in previous_components:
+                if c.name:
+                    # TODO improve efficiency here from the for loop, but the following commented out code doesn't work for module components since they have names but aren't /in/ the module so not named directly there.
+                    # current_c = getattr(current_graph, c.name)
+                    current_c = [comp for comp in current_components if comp.name == c.name][0]
+                    component_map[c.uuid] = current_c.uuid
+                    current_level[c.uuid] = current_c.uuid
+
```

Comment:
The above implementation scale quadratically with the name of components. It can be changed to be linear with the following codes:
```py
name_pre  = {name: c for c in previous_components if c.name}
name_cur  = {name: c for c in current_components if c.name}
...
```